### PR TITLE
Add in helper functionality to guzzlr

### PR DESCRIPTION
**Pull request by pstalcup**
Just some small guzzlr helpers. Cheapest platinum cocktail assumed you have a way to craft it for free, so I added a way to use mall price of the completed cocktail instead.

### CODE REVIEW

File: src/resources/2020/Guzzlr.ts [hunk 1]
```diff
@@ -198,9 +198,26 @@ export const platinumCocktailToIngredient = invertMap(
...
+    return (maxBy(
+      Array.from(ingredientToPlatinumCocktail),
+      (ingredientAndCocktail: [Item, Item]) =>
+        mallPrice(ingredientAndCocktail[0])
+    ) ?? defaultCocktail)[1];
+  } else {
+    return (maxBy(
+      Array.from(ingredientToPlatinumCocktail),
+      (ingredientAndCocktail: [Item, Item]) =>
+        mallPrice(ingredientAndCocktail[1])
```
Comment (horrible-little-slime) [inline src/resources/2020/Guzzlr.ts:213]:
```suggestion
        Math.min(mallPrice(ingredientAndCocktail[0]), mallPrice(ingredientAndCocktail[1]))
```

Comment (horrible-little-slime) [reply src/resources/2020/Guzzlr.ts:213]: Just because we might as well

**Review by horrible-little-slime — APPROVED** [review]

File: src/resources/2020/Guzzlr.ts (hunk 1 above)
Comment (pstalcup) [reply src/resources/2020/Guzzlr.ts:213]: Resolved this even more elegantly

File: src/resources/2020/Guzzlr.ts [hunk 2]
```diff
@@ -198,13 +198,13 @@ export const platinumCocktailToIngredient = invertMap(
...
 
-export function getCheapestPlatinumCocktail(freeCraft: boolean = true): Item {
+export function getCheapestPlatinumCocktail(freeCraft = true): Item {
   const defaultCocktail = [$item`Dish of Clarified Butter`, $item`Buttery Boy`];
   if (freeCraft) {
     return (maxBy(
       Array.from(ingredientToPlatinumCocktail),
       (ingredientAndCocktail: [Item, Item]) =>
-        mallPrice(ingredientAndCocktail[0])
+        Math.min(...ingredientAndCocktail.map(mallPrice))
```
Comment (horrible-little-slime) [inline src/resources/2020/Guzzlr.ts:207]: I would test this--ASH functions don't know how to handle being given extra inputs, and so it'll try to interpret it as mallPrice(element, index, array) and have a bad time

Comment (horrible-little-slime) [reply src/resources/2020/Guzzlr.ts:207]:
```suggestion
        Math.min(...ingredientAndCocktail.map((item) => mallPrice(item)))
```

**Review by gausie — APPROVED** [review]

**Pull request merged**